_posts/2021-02-11-tabnet/tabnet.Rmd

```diff
@@ -0,0 +1,411 @@
+---
+title: "torch, tidymodels, and high-energy physics"
+description: > 
+ Today we introduce tabnet, a torch implementation of "TabNet: Attentive Interpretable Tabular Learning" that is fully integrated with the tidymodels framework. Per se, already, tabnet was designed to require very little data pre-processing; thanks to tidymodels, hyperparameter tuning (so often cumbersome in deep learning) becomes convenient and even, fun!
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatabnet
+date: 02-11-2021
+bibliography: bibliography.bib
+categories:
+  - Torch
+  - R
+  - Tabular Data
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/d.jpg
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+So what's with the clickbait (*high-energy physics*)? Well, it's not just clickbait only. To showcase TabNet, we will be using the [Higgs](https://archive.ics.uci.edu/ml/datasets/HIGGS) dataset (@higgs) available at UCI Machine Learning Repository. I don't know about you, but I always enjoy using datasets that motivate me to learn more about things. But first, let's get acquainted with the main actors of this post!
+
+# TabNet
+
+TabNet was introduced in @arik2020tabnet. It is interesting for three reasons:
+
+-   It claims highly competitive performance on tabular data, an area where deep learning has not gained so much of a reputation yet.
+
+-   It includes interpretability[^1] features by design.
+
+-   It is claimed to significantly profit from self-supervised pre-training, again in an area where this is all but undeserving of mention.
```

Comment:
> an area where this is all but undeserving of mention.

I'm not totally sure what you mean by this. Are you saying it's not worth mentioning? If you mention it, don't say it's not worth mentioning—you did, and it's good! Own it! 💃

Reply:
I think I'm experimenting with how far I could go chaining negations :-)))

Ooooh wait ... no ... I'm not! I've been so confused about this I even talked with my husband about it, but I' STILL doing it wrong: "all but" for me always means what it would mean in German: "everything but not x" ... 
I will never learn that

Reply:
oh, realizing it should probably be _anything but_